cmd/frontend/graphqlbackend/code_host_mutation.go

```
@@ -0,0 +1,66 @@
+package graphqlbackend
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/graph-gophers/graphql-go"
+	"github.com/pkg/errors"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/backend"
+	"github.com/sourcegraph/sourcegraph/internal/db"
+)
+
+func (r *schemaResolver) SetCodeHostRepos(ctx context.Context, args struct {
+	ID       graphql.ID
+	Repos    *[]string
+	AllRepos bool
+}) (*EmptyResponse, error) {
+	id, err := unmarshalExternalServiceID(args.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	es, err := db.ExternalServices.GetByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	// 🚨 SECURITY: make sure the user is either site admin or the same user being requested
+	if err := backend.CheckSiteAdminOrSameUser(ctx, es.NamespaceUserID); err != nil {
+		return nil, err
+	}
+
+	cfg, err := es.Configuration()
+	if err != nil {
+		return nil, err
+	}
+
+	ra, ok := cfg.(extsvc.ReposSetter)
+	if !ok {
+		return nil, errors.Errorf("Code host %s (kind %s) does not implement extsvc.RepoAdder", args.ID, es.Kind)
+	}
+
+	var repos []string
+	if args.Repos != nil {
+		repos = *args.Repos
+	}
+	err = ra.SetRepos(args.AllRepos, repos)
+	if err != nil {
+		return nil, err
+	}
+
+	buf, err := json.MarshalIndent(cfg, "", "  ")
```

This error is not checked, is that on purpose?